sklearn/preprocessing/tests/test_data.py

```diff
@@ -191,19 +210,269 @@ def test_scaler_2d_arrays():
+            scaler_incr = scaler_incr.partial_fit(X[batch])
+
+        assert_array_almost_equal(scaler_batch.mean_, scaler_incr.mean_)
+        assert_equal(scaler_batch.var_, scaler_incr.var_)  # Nones
+        assert_equal(scaler_batch.n_samples_seen_, scaler_incr.n_samples_seen_)
+
+        # Test std after 1 step
+        batch0 = slice(0, chunk_size)
+        scaler_incr = StandardScaler().partial_fit(X[batch0])
+        if chunk_size == 1:
+            assert_array_almost_equal(np.zeros(n_features, dtype=np.float64),
+                                      scaler_incr.var_)
+            assert_array_almost_equal(np.ones(n_features, dtype=np.float64),
+                                      scaler_incr.scale_)
+        else:
+            assert_array_almost_equal(np.var(X[batch0], axis=0),
+                                      scaler_incr.var_)
+            assert_array_almost_equal(np.std(X[batch0], axis=0),
+                                      scaler_incr.scale_)  # no constants
+
+        # Test std until the end of partial fits, and
+        scaler_batch = StandardScaler().fit(X)
+        scaler_incr = StandardScaler()  # Clean estimator
+        for i, batch in enumerate(gen_batches(n_samples, chunk_size)):
+            scaler_incr = scaler_incr.partial_fit(X[batch])
+            if batch.stop != n_samples:
+                assert_equal((i + 1) * chunk_size, scaler_incr.n_samples_seen_)
+            else:
+                assert_equal(i * chunk_size + (batch.stop - batch.start),
+                             scaler_incr.n_samples_seen_)
+
+        assert_array_almost_equal(scaler_batch.var_, scaler_incr.var_)
+        assert_equal(scaler_batch.n_samples_seen_, scaler_incr.n_samples_seen_)
+
+    # Test with 1D list
+    X = [[0., 1., 2, 0.4, 1.]]
+    scaler_incr = StandardScaler().partial_fit(X)
+    assert_array_almost_equal(np.array([0., 1., 2, 0.4, 1.]),
+                              scaler_incr.mean_)
+    assert_array_almost_equal(np.zeros(5, dtype=np.float64),
+                              scaler_incr.var_)
+    assert_array_almost_equal(np.ones(5, dtype=np.float64),
+                              scaler_incr.scale_)
+    assert_equal(1, scaler_incr.n_samples_seen_)
+
+
+def test_standard_scaler_partial_fit_numerical_stability():
+    # Test if the incremental computation introduces significative errors
+    # for large datasets with values of large magniture
+    rng = np.random.RandomState(0)
+    n_features = 2
+    n_samples = 100
+    offsets = rng.uniform(-1e15, 1e15, size=n_features)
+    scales = rng.uniform(1e3, 1e6, size=n_features)
+
+    chunk_size = 1
+    data_chunks = [rng.randn(chunk_size, n_features) * scales + offsets
+                   for i in range(n_samples)]
+
+    X = np.vstack(data_chunks)
+    scaler_batch = StandardScaler().fit(X)
+
+    scaler_incr = StandardScaler()
+    for i, data_chunk in enumerate(data_chunks):
+        scaler_incr = scaler_incr.partial_fit(data_chunk)
```

Comment:
nit: You could also do

    X = rng.randn(n_samples, n_features) * scales + offsets
    for data_chunk in X:
         
saves a couple of lines of code.